src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Runtime/SpecifyIFormatProvider.cs

```diff
@@ -201,6 +202,26 @@ public override void Initialize(AnalysisContext analysisContext)
                                     oaContext.ContainingSymbol.ToDisplayString(SymbolDisplayFormats.ShortSymbolDisplayFormat),
                                     correctOverload.ToDisplayString(SymbolDisplayFormats.ShortSymbolDisplayFormat)));
                         }
+
+                        if (targetMethod.Parameters.Length == 2 && targetMethod.Parameters[1].RefKind == RefKind.Out)
+                        {
+                            correctOverload = methodsWithSameNameAsTargetMethod.FirstOrDefault(m =>
+                                m.Parameters.Length == 4 &&
+                                m.Parameters[0].Type.Equals(targetMethod.Parameters[0].Type) &&
+                                string.Equals(m.Parameters[1].Type.Name, "NumberStyles", StringComparison.InvariantCulture) &&
+                                m.Parameters[2].Type.Equals(iformatProviderType) &&
+                                m.Parameters[3].Type.Equals(targetMethod.Parameters[1].Type) &&
+                                m.Parameters[3].RefKind == RefKind.Out);
+
+                            if (correctOverload != null)
+                            {
+                                oaContext.ReportDiagnostic(
+                                    invocationExpression.Syntax.CreateDiagnostic(IFormatProviderAlternateRule,
+                                    targetMethod.ToDisplayString(SymbolDisplayFormats.ShortSymbolDisplayFormat),
```

Comment:
Nit: could you either increase the indent for the 3 parameters or move the `invocationExpression` to the previous line. At first glance, it looks like the `ReportDiagnostic` is taking 4 parameters.